master/pkg/searcher/adaptive_asha.go

```diff
@@ -44,6 +44,9 @@ func getBracketMaxConcurrentTrials(
 	remainder := 0
 	numBrackets := len(maxTrials)
 	bracketMaxConcurrentTrials := make([]int, 0, numBrackets)
+	// Without this, the remainder will be less than numBrackets and later brackets will,
+	// not receive a constraint on bracketMaxConcurrentTrials.
+	maxConcurrentTrials = min(maxConcurrentTrials, numBrackets)
```

Comment:
i don't think this fixes it.  we allow users to pass in 0 if they want a reasonable default for max_concurrent_trials.  
i think we should do:
maxConcurrentTrials = max(numBrackets, maxConcurrentTrials) after line 49 
before we do
minTrials = maxConcurrentTrials / numBrackets

Reply:
🤦

Reply:
brain broke, thanks @liamcli